Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -573,30 +564,28 @@ extension RouteController: Router {
             }
         }
         
-        delegate?.router(self, willRerouteFrom: location)
-        NotificationCenter.default.post(name: .routeControllerWillReroute, object: self, userInfo: [
-            NotificationUserInfoKey.locationKey: location,
-        ])
+        announceImpendingReroute(at: location)
         
         self.lastRerouteLocation = location
         
         // Avoid interrupting an ongoing reroute
         if isRerouting { return }
         isRerouting = true
         
-        getDirections(from: location, along: progress) { [weak self] (session, result) in
+        calculateRoutes(from: location, along: progress) { [weak self] (session, result) in
             self?.isRerouting = false
             
             guard let strongSelf: RouteController = self else {
                 return
             }
             
             switch result {
-            case let .success(response):
-                guard let route = response.routes?.first else { return }
+            case let .success(indexedResponse):
+                let response = indexedResponse.routeResponse
+                guard let route = response.routes?[indexedResponse.routeIndex] else { return }
                 guard case let .route(routeOptions) = response.options else { return } //TODO: Can a match hit this codepoint?
-                strongSelf._routeProgress = RouteProgress(route: route, options: routeOptions, legIndex: 0)
-                strongSelf._routeProgress.currentLegProgress.stepIndex = 0
+                strongSelf.routeProgress = RouteProgress(route: route, options: routeOptions)
+                strongSelf.indexedRouteResponse = indexedResponse
```

Comment:
The route index and step index default to 0, so I removed the explicit call to change the step index to 0.